### ISSUE TITLE
Game hangs on start of following turn if drawn card is played before draw animation completes 

### ISSUE THREAD

**Issue by Xinhuan** [HearthSim/hs-bugs#324]
https://www.youtube.com/watch?v=eyoOrOttQWQ

The start of the video shows a failed attempt at causing the hang by waiting to play the card. The actual bug starts at the 0:40 in the video.

To achieve the bug, when you hear the sound effect for "Your Turn", immediately click once at the position where the drawn card would be, move the mouse out of the hand, and then click to play the card as soon as the cards in hand turn green (about slightly less than 1 second between the 2 clicks).

When your next turn starts, the card draw animation does not start properly, resulting in a game animation hang.

This is reproduced by Culinko quite a number of times, and then by me for the 1080p video.

**Comment by Patashu** [comment]
The bug that leads to this hang (being able to grab a card before the drawing animation finishes) should have its own issue, IMO.

A gif showing just that (but the start of the video also shows it): http://gph.is/1WzCrJV

**Comment by Patashu** [comment]
https://www.youtube.com/watch?v=PdcovHDMaQU&t=1m53s is related to this or perhaps #350

**Comment by Patashu** [comment]
https://www.reddit.com/r/hearthstone/comments/3di6ss/weird_bug_in_tavern_brawl/ is related?

**Comment by culinko** [comment]
So I was trying to reproduce the report from the last link and I realized that it's from Tavern Brawl: Encounter at the Crossroads (by the date of the post). It's the brawl where the basic deck was replaced by 30 random cards at the start of the match and sometimes it also visually showed the basic cards being removed. I believe this no longer works this way, because a) I can't remember it happening at the last Encounter at the Crossroads and b) basic decks are now editable. So I'm pretty sure even if we get this brawl again, the bug is already fixed. But if you remind me then, I can still test this.
